### PR TITLE
Add embedding finetune demo

### PR DESCRIPTION
**Pull request by KPatr1ck**

### CODE REVIEW

File: demo/text_classification/embedding/embedding_classification.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            correct = self.metric.compute(probs, labels)
+            acc = self.metric.update(correct)
+            return probs, loss, {'acc': acc}
+        return probs
+
+
+if __name__ == '__main__':
+    embedder = hub.Module(name=args.hub_embedding_name)
+    vocab = embedder.vocab.token_to_idx
+    tokenizer = CustomTokenizer(vocab=vocab, cut_function=jieba.cut)
```
Comment (Steffy-zxf) [inline demo/text_classification/embedding/embedding_classification.py]: tokenizer 换成`JiebaTokenizer`，保持和 paddlenlp 一致。直接调用paddlenlp.JiebaTokenizer即可
这里换成 `module.get_vocab_path()`, 保持和Transformer类模型使用统一。

File: paddlehub/text/tokenizer.py [hunk 2]
```diff
@@ -133,11 +138,14 @@ def tokenize(self, text: str):
...
         '''
         if self.tokenize_chinese_chars:
-            splitted_tokens = self.cut_function(text=text)
+            splitted_tokens = self.cut_function(text)
         else:
-            splitted_tokens = self.basic_tokenizer.tokenize(text=text)
+            splitted_tokens = self.basic_tokenizer.tokenize(text)
         return splitted_tokens
 
+    def __call__(self, text):
```
Comment (Steffy-zxf) [inline paddlehub/text/tokenizer.py]: 为什么 要增加call方法

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
1. word embedding module 需要增加get_vocab_path 方法
2. tokenizer直接使用paddlenlp.JiebaTokenizer
3. 增加使用embedding 预测脚本

File: paddlehub/text/tokenizer.py (hunk 2 above)
Comment (KPatr1ck) [reply paddlehub/text/tokenizer.py]: 已回退，CustomTokenizer不修改
用paddlenlp的JiebaTokenizer

File: demo/text_classification/embedding/embedding_classification.py (hunk 1 above)
Comment (KPatr1ck) [reply demo/text_classification/embedding/embedding_classification.py]: tokenizer已换成JiebaTokenizer

File: paddlehub/datasets/base_nlp_dataset.py [hunk 3]
```diff
@@ -313,10 +354,15 @@ def _convert_examples_to_records(self, examples: List[InputExample]) -> List[dic
                 tokenizer=self.tokenizer,
                 tokens=example.text_a.split(self.split_char),
                 labels=example.label.split(self.split_char))
-            record = self.tokenizer.encode(
-                text=tokens, max_seq_len=self.max_seq_len)
-            # CustomTokenizer will tokenize the text firstly and then lookup words in the vocab
-            # When all words are not found in the vocab, the text will be dropped.
+            if isinstance(self.tokenizer, PretrainedTokenizer):
+                record = self.tokenizer.encode(text=tokens, max_seq_len=self.max_seq_len)
+            elif isinstance(self.tokenizer, JiebaTokenizer):
```
Comment (Steffy-zxf) [inline paddlehub/datasets/base_nlp_dataset.py:369]: 序列标注数据集，应该不需要切词。因为数据集已经划分好了的每个token

File: paddlehub/utils/utils.py [hunk 4]
```diff
@@ -367,3 +368,19 @@ def reseg_token_label(tokenizer, tokens: List[str], labels: List[str] = None):
             if len(sub_token) < 2:
                 continue
         return ret_tokens, None
+
+
+def normalize_sequence(ids: List[int], max_seq_len: int, vocab: Vocab, return_length: bool = True):
```
Comment (Steffy-zxf) [inline paddlehub/utils/utils.py]: 建议拆分成2个函数，一个pad_sequence，一个trunc_sequence。

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]

File: demo/text_classification/embedding/embedding_classification.py (hunk 1 above)
Comment (KPatr1ck) [reply demo/text_classification/embedding/embedding_classification.py]: tokenizer统一为从model中获取

File: paddlehub/datasets/base_nlp_dataset.py (hunk 3 above)
Comment (KPatr1ck) [reply paddlehub/datasets/base_nlp_dataset.py:369]: 针对序列标注数据集，以分隔符中间的token为最小单位转换为词典id

File: paddlehub/utils/utils.py (hunk 4 above)
Comment (KPatr1ck) [reply paddlehub/utils/utils.py]: 已拆开成pad_sequence和trunc_sequence。

File: paddlehub/utils/utils.py [hunk 5]
```diff
@@ -367,3 +367,23 @@ def reseg_token_label(tokenizer, tokens: List[str], labels: List[str] = None):
             if len(sub_token) < 2:
                 continue
         return ret_tokens, None
+
+
+def pad_sequence(ids: List[int], max_seq_len: int, pad_token_id: int):
+    '''
+    Pad s sequence to max_seq_len
```
Comment (Steffy-zxf) [inline paddlehub/utils/utils.py]: Pads

File: paddlehub/utils/utils.py [hunk 6]
```diff
@@ -367,3 +367,23 @@ def reseg_token_label(tokenizer, tokens: List[str], labels: List[str] = None):
...
+    assert len(ids) <= max_seq_len, \
+        f'The input length {len(ids)} is greater than max_seq_len {max_seq_len}. '\
+        'Please check the input list and max_seq_len if you really want to pad a sequence.'
+    return ids[:] + [pad_token_id]*(max_seq_len-len(ids))
+
+
+def trunc_sequence(ids: List[int], max_seq_len: int):
+    '''
+    Truncate a sequence to max_seq_len
+    '''
```
Comment (Steffy-zxf) [inline paddlehub/utils/utils.py:385]: Truncates

File: demo/text_classification/embedding/model.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        encoded_text = paddle.tanh(summed)
+
+        # Shape: (batch_size, hidden_size)
+        fc1_out = paddle.tanh(self.fc1(encoded_text))
+        # Shape: (batch_size, fc_hidden_size)
+        fc2_out = paddle.tanh(self.fc2(fc1_out))
+        # Shape: (batch_size, num_classes)
+        logits = self.output_layer(fc2_out)
+
+        probs = F.softmax(logits, axis=1)
```
Comment (Steffy-zxf) [inline demo/text_classification/embedding/model.py:117]: CrossEntropy里面 会自动加上 softmax。
所以训练时，不需要计算F.softmax，但预测时需要。

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]

File: paddlehub/utils/utils.py (hunk 5 above)
Comment (KPatr1ck) [reply paddlehub/utils/utils.py]: 已改正

File: paddlehub/utils/utils.py (hunk 6 above)
Comment (KPatr1ck) [reply paddlehub/utils/utils.py:385]: 已改正

**Review by Steffy-zxf — APPROVED** [review]
LGTM
